### PR TITLE
CI: reuse workflow

### PR DESCRIPTION
**Pull request by Borda**
## What does this PR do?

Fixes #\<issue_number>

## Before submitting

- [ ] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure to update the docs?
- [x] Did you write any new necessary tests?

## PR review

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?

Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
I want to use the reusable workflow in our CI but seems it does not work for PR. I have followed the example in docs:
https://docs.github.com/en/actions/using-workflows/reusing-workflows#using-outputs-from-a-reusable-workflow
and the only change I made is to update the triggers in the second workflow to be:
```yaml
on:
  - push
  - pull_request
```
with the following error message:
```
Invalid workflow file : .github/workflows/ttt2.yml#L1
The workflow is not valid. PyTorchLightning/metrics/.github/workflows/ttt1.yml@fa65582b43a1f7105fe98e9b6d41b210747ebab4: Expected document st
```
in fact, for the same commit, the action triggered by push pass and the one with PR fails
image

**Comment by Borda** [comment]
an alternative could be using composite steps: [outputs for composite actions](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#outputs-for-composite-actions)

**Comment by Borda** [comment]
it was admitted as an issue on the GH side and will be taken care of it...

**Comment by Borda** [comment]
Just got message that this issue was fixed and seems to be working correctly now

File: .github/workflows/ci_test-full.yml [hunk 1]
```diff
@@ -62,6 +42,8 @@ jobs:
     timeout-minutes: 95
 
     steps:
+    # fixme
+    #- run: echo "${{ needs.check-diff.outputs.test-dirs }}"
```
Comment (Borda) [inline .github/workflows/ci_test-full.yml:46]:
```suggestion
```

File: .github/workflows/ci_test-conda.yml [hunk 2]
```diff
@@ -99,16 +79,18 @@ jobs:
         pip install -q "numpy==1.20.0"  # try to fix cocotools for PT 1.4 & 1.9
         # import of PILLOW_VERSION which they recently removed in v9.0 in favor of __version__
         pip install -q "Pillow<9.0"  # It messes with torchvision
-        pip install --requirement requirements.txt --quiet
+        # pip install --requirement requirements.txt --quiet
```
Comment (justusschock) [inline .github/workflows/ci_test-conda.yml:82]:
```suggestion
```

File: .github/workflows/ci_test-conda.yml [hunk 3]
```diff
@@ -99,16 +79,18 @@ jobs:
...
         python -c "from torch import __version__ as ver; assert '.'.join(ver.split('.')[:2]) == '${{ matrix.pytorch-version }}', ver"
 
     - name: Testing
       env:
-        TEST_DIRS: "${{ needs.check-diff.outputs.focus }}"
+        TEST_DIRS: "${{ needs.check-diff.outputs.test-dirs }}"
       run: |
-        # NOTE: run coverage on tests does not propagare faler status for Win, https://github.com/nedbat/coveragepy/issues/1003
+        phmdoctest README.md --outfile tests/test_readme.py
+        # NOTE: run coverage on tests does not propagare faler status for Win,
```
Comment (justusschock) [inline .github/workflows/ci_test-conda.yml:92]:
```suggestion
        # NOTE: run coverage on tests does not propagate failure status for Win,
```

File: .github/workflows/ci_test-conda.yml [hunk 4]
```diff
@@ -99,16 +79,18 @@ jobs:
...
         pip list
         python -c "from torch import __version__ as ver; assert '.'.join(ver.split('.')[:2]) == '${{ matrix.pytorch-version }}', ver"
 
     - name: Testing
       env:
-        TEST_DIRS: "${{ needs.check-diff.outputs.focus }}"
+        TEST_DIRS: "${{ needs.check-diff.outputs.test-dirs }}"
       run: |
-        # NOTE: run coverage on tests does not propagare faler status for Win, https://github.com/nedbat/coveragepy/issues/1003
+        phmdoctest README.md --outfile tests/test_readme.py
```
[the remark this answers is not in the document]
Comment (Borda) [reply .github/workflows/ci_test-conda.yml:91]: it generated tests from examples in readme so all are properly tested

**Review by ethanwharris — APPROVED** [review]
LGTM

**Pull request merged**